src/pages/get-premium/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { v4 as uuidv4 } from "uuid";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import useUserStore from "@/stores/user";
+import useToastStore from "@/stores/toast";
+import BaseButton from "@/components/base/BaseButton";
+import LoadingSpinner from "@/components/LoadingSpinner";
+import { createCheckOutSessionService } from "@/services/payments";
+
+export default function GetPremiumPage() {
+  const router = useRouter();
+  const { profile, getPremiumStatus, isLoading } = useUserStore();
+  const addToast = useToastStore((state) => state.addToast);
+  const [isCheckoutSessionLoading, setIsCheckoutSessionLoading] =
+    useState(false);
+
+  async function handleClickSubscribe(type: "monthly" | "yearly") {
+    // Get user to sign in
+    if (!profile) {
+      router.push("/signin/premium");
+      return;
+    }
+
+    // Check if user is already a premium user
+    if (getPremiumStatus()) {
+      addToast({
+        uniqueId: uuidv4(),
+        className: "bg-green-50 text-green-700",
+        content: "You are already a premium user!",
+      });
```

Comment:
create `addSuccessToast` function with preconfigured defaults. Same for `addErrorToast`